modules/webstat/lib/webstat_engine.py

```diff
@@ -1886,15 +1886,14 @@ def create_custom_summary_graph(data, path, title):
     """
     # If no input, we don't bother about anything
     if len(data) == 0:
-        return
+        return False
     os.environ['HOME'] = CFG_TMPDIR
 
-    try:
-        import matplotlib
-        matplotlib.use('Agg')
-        import matplotlib.pyplot as plt
-    except ImportError:
-        return
+
+    import matplotlib
+    matplotlib.use('Agg')
+    import matplotlib.pyplot as plt
+
     # make a square figure and axes
```

Comment:
matplotlib is only recommended (in the INSTALL file), not mandatory.  Since it is only use for this minor functionality, I think it is good to keep try/except.  But you can surely put it to requirements for those installations that use pip.  This would make graph OK, but would not break other installations, or ask them to install it.  (Kind of defensive programming.)  Can you please amend your PR in this respect?